ui/helpers/object.js

```diff
@@ -21,3 +20,6 @@ export const extendProperties = function extendProperties(dest, src) {
   }
   return dest;
 };
+
+export const didObjectsChange = (firstObj, secondObj, fields) =>
```

Comment:
I was getting sick of long ``shouldComponentUpdate`` compares and thought this would be cleaner.  But my spider sense tells me that you'll come back with, "Didn't you know about ``neatoCompare``?  It's way better..."

Reply:
I'm wondering if where this is being used, you could instead use a PureComponent since it implements a shallow comparison of props by default whenever props change (or state)?

Reply:
Yeah, this is a great question  The problem comes from those contexts that I wrote.  With a ``PureComponent``, it thinks things like the functions as well as ``push`` and a few others they provide are getting updated.  So it will do tons of re-renders without cause.  This use of ``shouldComponentUpdate`` is necessary until I re-work those into either Redux or some other mechanism that is more effective.  

Boy, those specialized HOC seemed like a good idea at the time.  It sure simplified the code, but cause all kinds of perf problems.  Lessons learned...  :)

Reply:
Well Redux uses the Context API under the hood so I don't know how much help that will be (plus, weren't we planning to remove Redux entirely since the only thing using it is a small part of the app)? If you're interested in some Redux-like features, there is a new `useReducer` hook but I don't know how helpful/relevant that will be. I haven't looked into hooks too much in depth since `useEffects` seems like a bit of a paradigm shift.